dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/SuperClassSymbol.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+/**
+ * Symbol for a super class. This symbol is a special case of {@link ClassSymbol} when it is
+ * referenced only from its subclasses. Treating super class symbols apart from {@link ClassSymbol}
```

Comment:
What happens when a class is both a superclass and referenced directly from a non-subclass?

Reply:
- The reference to the class from subclass is validated according to `final` keyword.
- The references to the class from both subclass and non-subclass are validated according to class availability (`CLASS_NOT_FOUND` or not) and accessibility (`public`, `protected`, etc.).

Added a mention to 'final' keyword.